plugin/trino-hive/src/main/java/io/trino/plugin/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -419,6 +419,40 @@ private static TableStatistics getTableStatistics(
         return result.build();
     }
 
+    private static TableStatistics createEmptyTableStatisticsWithPartitionColumnStatistics(
+            Map<String, ColumnHandle> columns,
+            Map<String, Type> columnTypes,
+            List<HivePartition> partitions)
+    {
+        TableStatistics.Builder result = TableStatistics.builder();
+        for (Map.Entry<String, ColumnHandle> column : columns.entrySet()) {
+            HiveColumnHandle columnHandle = (HiveColumnHandle) column.getValue();
+            if (columnHandle.isPartitionKey()) {
+                result.setColumnStatistics(
+                        columnHandle,
+                        createPartitionColumnStatistics(columnHandle, columnTypes.get(column.getKey()), partitions));
+            }
+        }
+        return result.build();
+    }
+
+    private static ColumnStatistics createPartitionColumnStatistics(HiveColumnHandle column, Type type, List<HivePartition> partitions)
+    {
+        if (partitions.isEmpty()) {
+            return ColumnStatistics.empty();
+        }
+
+        double estimatedNullsCount = partitions.stream()
+                .filter(partition -> partition.getKeys().get(column).isNull())
+                .count();
```

Comment:
`calculateDistinctPartitionKeys` already gives us non-nulls count, so we can get nulls counts without iterating over partitions again by `partitions.size() - ndv)`

Reply:
We are taking the assumption that each partition has the same number of rows. So, we have to calculate non-null values considering all available partitions (multi-column partition). Therefore, we can't rely on `calculateDistinctPartitionKeys` since it gives a distinct count.

Reply:
Right, I missed the "distinct" in that method.